File: cycleLife/app/src/main/java/com/example/ejercicio_1/MainActivity.java
package com.example.ejercicio_1;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    private static final String tag  = "cycle";

    TextView txt_v;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Toast.makeText(this,"_onCreate",Toast.LENGTH_SHORT).show();
        txt_v = findViewById(R.id.txt_hola);
        Log.d(tag,"_onCreate");

    }
    @Override
    protected void  onPause(){
        super.onPause();
        Toast.makeText(this,"_onPause",Toast.LENGTH_SHORT).show();
        Log.d(tag,"_onPause");
    }

    @Override
    protected void onStart(){
        super.onStart();
        Toast.makeText(this,"_onStart",Toast.LENGTH_SHORT).show();
        Log.d(tag,"_onStart");
        //txt_v.setText("Activa");
    }

    @Override
    protected void onResume(){
        super.onResume();
        Toast.makeText(this,"_onResume",Toast.LENGTH_SHORT).show();
        Log.d(tag,"_onResume");
    }

    @Override
    protected void onStop(){
        super.onStop();
        Toast.makeText(this,"_onStop",Toast.LENGTH_SHORT).show();
        Log.d(tag,"_onStop");
        txt_v.setText("Fuera de foco");
    }

    @Override
    protected void onRestart(){
        super.onRestart();
        Toast.makeText(this,"_onRestart",Toast.LENGTH_SHORT).show();
        Log.d(tag,"_onRestart");
    }

    @Override
    protected void onDestroy(){
        super.onDestroy();
        Toast.makeText(this,"_onDestroy",Toast.LENGTH_SHORT).show();
        Log.d(tag,"_onDestroy");
    }
}

File: AppFoof/app/src/main/java/chab/dog/appfoof/activities/MainActivity.kt
package chab.dog.appfoof.activities

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.ImageView
import androidx.fragment.app.Fragment
import chab.dog.appfoof.R
import chab.dog.appfoof.fragments.HomeFragment
import chab.dog.appfoof.fragments.ProfileFragment
import chab.dog.appfoof.fragments.SettingFragment
import com.google.android.material.bottomnavigation.BottomNavigationView
import java.util.*

class MainActivity : AppCompatActivity() {
    lateinit var imgDice : ImageView
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val navMenu : BottomNavigationView = findViewById(R.id.navMenu)
        navMenu.setOnNavigationItemSelectedListener {
                menuItem ->
            when (menuItem.itemId) {
                R.id.homeFragment -> {
                    val fragment = HomeFragment.newInstance()
                    openFragment(fragment)
                    true
                }
                R.id.profileFragment -> {
                    val fragment = ProfileFragment.newInstance()
                    openFragment(fragment)
                    true
                }
                R.id.settingFragment -> {
                    val fragment = SettingFragment.newInstance()
                    openFragment(fragment)
                    true
                }
                else -> false
            }
        }
    }

    private fun openFragment(fragment: Fragment) {
        val transaction = supportFragmentManager.beginTransaction()
        transaction.replace(R.id.fragment, fragment)
        transaction.addToBackStack(null)
        transaction.commit()
    }
}
